### PR TITLE
Added todo example

### LINKED ISSUE #57: Create the stereotypical "todo" example

### PR DESCRIPTION
**Pull request by edfloreshz**
I added a simple todo example, this is still in draft because I haven't figured out how to modify the database in the `on_key` event. 

I could use some guidance here. 🙏🏼 

Closes #57

### CODE REVIEW

**Comment by ecton** [comment]
Sorry for the delay in getting back to you! Thank for taking a stab at this. I had previously [started one](https://github.com/khonsulabs/gooey/blob/02d6b343f1320724830067fcf73ce8d71da0bb99/examples/todo.rs) a long time ago, then I removed it before releasing v0.1 because I never finished it. That being said, however, it does have the basic "Add a task" flow hooked up through a default button.

Why was it never finished? At the time I started it, I didn't have enough widgets to make it work the way I wanted. That inspired me to do some other work, but I just never got around to taking the example across the finish line.

In short, the data structure for the todos should be some sort of collection stored inside of a `Dynamic`. This can then be synchronized with a `Children` widget. This API *is not ideal* and is probably one of the ugliest parts of Gooey's design right now.

**Comment by hydra** [comment]
Just wondering if this PR is going to be updated?

Additionally, the TodoMVC website has an application specification for a 'Todo' application, which can be found here:

https://github.com/tastejs/todomvc/blob/master/app-spec.md#functionality

It would probably be useful if any TODO example had the same well-known functionality, similar to the effort in the 7GUIs requirements (https://eugenkiss.github.io/7guis/tasks)

**Comment by edfloreshz** [comment]
I'm quite busy with other projects but I'll try to find the time to update this.
